lsp/nls/src/requests/hover.rs

```diff
@@ -156,57 +160,47 @@ pub fn handle(
     if let Some(hover) = hover_data {
         let mut contents = Vec::new();
 
-        // If we can't determine a static type through the typechecker because we are outside of a
-        // statically typed block, but the term points to a definition with a type annotation, we
-        // use this annotation insted.
-        let mut type_annots: Vec<_> = hover
+        // Collect all the type and contract annotations we can find. (We don't distinguish between them.)
+        let mut annotations: Vec<_> = hover
             .metadata
             .iter()
-            .filter_map(|m| Some(m.annotation.typ.as_ref()?.typ.to_string()))
+            .flat_map(|m| {
+                let maybe_typ = m.annotation.typ.as_ref().map(|typ| typ.typ.to_string());
+                let contracts = m
+                    .annotation
+                    .contracts
+                    .iter()
+                    .map(|c| c.label.typ.to_string());
+                maybe_typ.into_iter().chain(contracts)
+            })
+            .chain(
+                hover
+                    .values
+                    .iter()
+                    .flat_map(annotated_contracts)
+                    .map(|contract| contract.label.typ.to_string()),
+            )
             .collect();
+        annotations.sort();
+        annotations.dedup();
```

Comment:
I'm not entirely sure it's ok to sort the annotations. The order of contracts is meaningful in Nickel (different order of application might give a different result). Additionally, if there's no static type, the first contract of the list is taken to be the static type, which also makes a difference statically, and not only at runtime.

Reply:
Ok, I'm deduping them without sorting now.